elastic/logstash
# Hythloday zero ecs compat default fix

- doc

## Release notes
[rn:skip]

## What does this PR do?

This PR updates the documentation for pipeline.ecs_compatibility setting to reflect the new default setting of 'v8' in Logstash 8.x.

## Why is it important/What is the impact to the user?

This PR clarifies a default setting changes between Logstash 7 and 8.  Users from Logstash 7 had pipeline.ecs_compatibility: disabled set as the default, as the v1 and v8 values were Beta features.  This can cause pipeline logic to no longer work as intended when the default in Logstash 8 is set to 'v8'.
By making users aware of this change via documentation, they are able to account for the setting either globally in the logstash.yml file or on a per-pipeline/filter processor basis.

## Checklist



~~- [ ] My code follows the style guidelines of this project~~
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x ] I have made corresponding changes to the documentation
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ] Check against 8.0 - 8.3.x to see when this fix should first appear..  Only version with the correct documentation to match this PR is from 8.4.  If this is the genesis of the default value change, then all versions from 8.5 on should be updated to reflect the change.

## How to test this PR locally

Create a Logstash 8 instance.  Create a pipeline with a filter phase that has the following conditional:
clone {
      clones => [ "compat-check" ]
    }
if [type] == "compat-check" {
  mutate {
     add_field => { "ecs_check" => "true"
  }
}

In the Input phase, simulate/read any document, and Output to Elasticsearch.

Using the default setting for pipeline.ecs_compatibility, you will receive 2 documents to the index, neither of which will contain the added field of 'ecs_check'.

Set pipeline.ecs_compatibility: disabled in the logstash.yml and restart the Logstash instance.  Run the document through the pipeline again, and you will have 2 documents in the index.  This time, one will have the 'ecs_check' field set to 'true' and the other doc will not have the field.

## Related issues


- 

## Use cases



## Screenshots



## Logs
